Source/Charts/Data/Implementations/Standard/ChartDataEntry.swift

```diff
@@ -108,32 +99,18 @@ open class ChartDataEntry: ChartDataEntryBase
     }
 }
 
-public func ==(lhs: ChartDataEntry, rhs: ChartDataEntry) -> Bool
-{
-    if lhs === rhs
-    {
-        return true
-    }
-    
-    if !lhs.isKind(of: type(of: rhs))
+// MARK: Equatable
+extension ChartDataEntry/*: Equatable*/ {
+    public static func ==(lhs: ChartDataEntry, rhs: ChartDataEntry) -> Bool
     {
-        return false
-    }
-    
-    if lhs.data !== rhs.data && !lhs.data!.isEqual(rhs.data)
-    {
-        return false
-    }
-    
-    if fabs(lhs.x - rhs.x) > Double.ulpOfOne
-    {
-        return false
-    }
-    
-    if fabs(lhs.y - rhs.y) > Double.ulpOfOne
-    {
-        return false
+        if lhs === rhs
+        {
+            return true
+        }
+
+        return lhs.isKind(of: type(of: rhs))
+            && (lhs.data?.isEqual(rhs.data) ?? true)
```

Comment:
for your second reply, yes it's bothering us (me) that we all have work to do and not having time to track the swift evolution. It could be lazy or just too many family errands. I am currently the only active person reviewing issues and PRs. But I am not swift expert and for those swift specific improvements, I need to find a time sit down and digest first. If it's small changes, I could merge for sure. But for very big changes, especially that will remove the support to objc usage, this gonna take time.

You are welcome on board to help us. You can try gitter or send emails to @danielgindi. We need to find out a solution for how to embrace swift better.